server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -129,6 +151,20 @@
             INDICES_RECOVERY_INTERNAL_LONG_ACTION_TIMEOUT_SETTING::get, TimeValue.timeValueSeconds(0),
             Property.Dynamic, Property.NodeScope);
 
+    /**
+     * recoveries would try to use files from available snapshots instead of sending them from the source node.
+     * defaults to `false`
+     */
+    public static final Setting<Boolean> INDICES_RECOVERY_USE_SNAPSHOTS_SETTING =
+        Setting.boolSetting("indices.recovery.use_snapshots", false, Property.NodeScope);
```

Comment:
I think we'll want this to be something that can be adjusted dynamically. I also wonder if it should be index-scoped rather than node-scoped.